devtools/mog/generate.go

```diff
@@ -172,20 +164,39 @@ func sourceFieldMap(fields []fieldConfig) map[string]fieldConfig {
 }
 
 type generated struct {
-	To   *ast.FuncDecl
-	From *ast.FuncDecl
+	To                     *ast.FuncDecl
+	PtrTo, ToPtr, PtrToPtr *ast.FuncDecl
+
+	From                         *ast.FuncDecl
+	PtrFrom, FromPtr, PtrFromPtr *ast.FuncDecl
 
 	// TODO: RoundTripTest *ast.FuncDecl
 }
 
+func (g *generated) Decls() []ast.Decl {
+	return []ast.Decl{
+		g.To,
+		g.PtrTo,
+		g.ToPtr,
+		g.PtrToPtr,
+		g.From,
+		g.PtrFrom,
+		g.FromPtr,
+		g.PtrFromPtr,
+	}
+}
+
 func generateToFunc(cfg structConfig, imports *imports) *ast.FuncDecl {
 	targetType := &ast.SelectorExpr{
 		X:   &ast.Ident{Name: path.Base(imports.AliasFor(cfg.Target.Package))},
 		Sel: &ast.Ident{Name: cfg.Target.Struct},
 	}
 
+	funcName := funcNameTo(cfg, false, false)
+
 	return &ast.FuncDecl{
-		Name: &ast.Ident{Name: funcNameTo(cfg)},
+		Doc:  newComments(funcName + " is a struct-to-struct conversion that translates FROM a protobuf"),
```

Comment:
I don't think we should hard code anything about protobuf here. The tool could be used to convert between `api` types and `structs` types, or `config` types and `structs` types, or used in other projects that have similar needs.

Reply:
removed the comments both for this reason and because now there aren't 8 functions per type